inventory_data.go

```diff
@@ -0,0 +1,162 @@
+// Copyright 2016 Mender Software AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"os"
+	"path"
+	"strings"
+	"syscall"
+
+	"github.com/mendersoftware/log"
+	"github.com/pkg/errors"
+)
+
+const (
+	inventoryToolPrefix = "mender-inventory-"
+)
+
+type InventoryDataGetter interface {
+	Get() (InventoryData, error)
+}
+
+func NewInventoryDataGetter(scriptsDir string) InventoryDataGetter {
+	return &InventoryDataRunner{
+		scriptsDir,
+		&osCalls{},
+	}
+}
+
+type InventoryDataRunner struct {
+	dir string
+	cmd Commander
+}
+
+func listRunnable(dpath string) ([]string, error) {
+	dp, err := os.Open(dpath)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to open %s", dpath)
+	}
+
+	finfos, err := dp.Readdir(0)
+	if err != nil {
+		// don't care about any FileInfo that were read up to this point
+		return nil, errors.Wrapf(err, "failed to readdir")
+	}
+
+	runnable := []string{}
+	for _, finfo := range finfos {
+		if !strings.HasPrefix(finfo.Name(), inventoryToolPrefix) {
+			continue
+		}
+
+		runBits := os.FileMode(syscall.S_IXUSR | syscall.S_IXGRP | syscall.S_IXOTH)
+		if finfo.Mode()&runBits == 0 {
+			continue
+		}
+
+		runnable = append(runnable, path.Join(dpath, finfo.Name()))
+	}
+
+	return runnable, nil
+}
+
+type tempInventoryAttribute []string
+
+func (tia tempInventoryAttribute) ToInventoryAttribute(name string) InventoryAttribute {
+	if len(tia) > 1 {
```

Comment:
`ToInventoryAttribute` just follows the convention of naming methods that are of some types 'public' API. Also keep in mind that all of this is part of 'main' package, hence it's not really exported. Renaming the method to `toInventoryAttribute` will not hide it from other files either.